### ISSUE TITLE
Feature request - Import/export of device config

### ISSUE THREAD

**Issue by kabili207** [zenaror/PicoAdapterGB#45]
This is a very low-priority request, and I'm not quite sure how it would best be implemented, but it would be really handy to have a way to backup the configuration from one adapter and restore it on a different one.

I'm designing two different PCBs for myself that use a replacement link ports instead of edge connectors (I believe I mentioned this to you recently in discord), one using the pinout shown in your other repo and one using stacksmashing's pinout, and it would be wonderful to not have to go through the registration process in mobile trainer for each board revision I make.

**Comment by zenaror** [comment]
Hi again🤣!

Yeah, I'm already thinking in something like this... this will probability need some extra software/script to do this, but it's totally possible.

The idea of the REON project, in the future, is to remove the needed to use the Mobile Trainer to make the Mobile Center config (this game able you to configure your login/email/password that other games uses, like the Trade Corner or the News Machine  in Pokemon Crystal)

I just forgot to open an issue for this, but since you opened, I appreciate 😁

**Comment by kabili207** [comment]
> The idea of the REON project, in the future, is to remove the needed to use the Mobile Trainer to make the Mobile Center config (this game able you to configure your login/email/password that other games uses, like the Trade Corner or the News Machine in Pokemon Crystal)

That would be amazing, especially since Mobile Trainer isn't really useful beyond configuring the adapter.

**Comment by zenaror** [comment]
> > The idea of the REON project, in the future, is to remove the needed to use the Mobile Trainer to make the Mobile Center config (this game able you to configure your login/email/password that other games uses, like the Trade Corner or the News Machine in Pokemon Crystal)
> 
> That would be amazing, especially since Mobile Trainer isn't really useful beyond configuring the adapter.

Exactly!
The problem is... how to do it 😅
For PC emulators and the usb implementation that Lorenzooone was making, it will be easy since both will use physical  files.... but here is a bit complicated... but not impossible

**Comment by Lorenzooone** [comment]
> This is a very low-priority request, and I'm not quite sure how it would best be implemented, but it would be really handy to have a way to backup the configuration from one adapter and restore it on a different one.
> 
> I'm designing two different PCBs for myself that use a replacement link ports instead of edge connectors (I believe I mentioned this to you recently in discord), one using the pinout shown in your other repo and one using stacksmashing's pinout, and it would be wonderful to not have to go through the registration process in mobile trainer for each board revision I make.

Now that saving has been updated for this project, dumping the save with my firmware and loading said save onto the other board with it will work as a temporary solution. The two firmwares' saves should be fully compatible with each other.

**Comment by zenaror** [comment]
Thw only config that shouldn't export for now is the wifi settings... any other config like custom server, relay token, etc... should be compatible as well

@Lorenzooone I was thinking in create a new command  in config that sends over Serial all configs... including the eeprom section that mobile trainer setup 
Same to import things... since the idea of reon later is to provide a pre configured file to eliminate the dependence of mobile trainer ...
What do you think?

**Comment by Lorenzooone** [comment]
> Thw only config that shouldn't export for now is the wifi settings... any other config like custom server, relay token, etc... should be compatible as well

True, because that is not part of the eeprom data. However, it won't overwrite it either, so you will be able to configure them once and save/load the rest of the data just fine.
 
> @Lorenzooone I was thinking in create a new command in config that sends over Serial all configs... including the eeprom section that mobile trainer setup Same to import things... since the idea of reon later is to provide a pre configured file to eliminate the dependence of mobile trainer ... What do you think?

It should be possible.
Another possibility in the future would be making a small GBA multibootable that does the setup.

**Comment by zenaror** [comment]
We are now able to dump the actual device eeprom

I'm working into the upload now
it will only missing the specific pico adapter info, like wifi settings, etc

but the libmobile standard, will work

**Comment by zenaror** [comment]
Apparently, the upload works as well ❤️

**Comment by zenaror** [comment]
Feature added in https://github.com/zenaror/PicoAdapterGB/commit/482a0aacd41e0927676648a1eb17323df47ad657
